File: backend/src/main/java/me/omartanner/modulepal/api/responses/body/module/ModuleMetadataResponseBody.java
package me.omartanner.modulepal.api.responses.body.module;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;
import me.omartanner.modulepal.api.dto.ModuleBasicData;
import me.omartanner.modulepal.api.dto.ModuleLeader;
import me.omartanner.modulepal.api.dto.ModuleReviewCount;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class ModuleMetadataResponseBody {
    private ModuleBasicData moduleBasicData;
    private List<String> academicYears;
    private List<ModuleLeader> leaders;
    private Long numReviews;
}

File: frontend/src/services/rest/responses/body/ModuleMetadataResponseBody.ts
import { ModuleBasicData } from "../../dto/ModuleBasicData";
import { ModuleLeader } from "../../dto/ModuleLeader";

export interface ModuleMetadataResponseBody {
    moduleBasicData: ModuleBasicData,
    academicYears: string[],
    leaders: ModuleLeader[],
    numReviews: number
}